src/PluginManager/Microsoft.Performance.Toolkit.PluginManager.Core.NuGet/NuGetPluginDiscoverer.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Performance.Toolkit.PluginManager.Core.Credential;
+using Microsoft.Performance.Toolkit.PluginManager.Core.Discovery;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+
+namespace Microsoft.Performance.Toolkit.PluginManager.Core.NuGet
+{
+    public sealed class NuGetPluginDiscoverer : IPluginDiscoverer
+    {
+        private readonly UriPluginSource pluginSource;
+        private readonly PackageSource packageSource;
+        private readonly Lazy<ICredentialProvider<UriPluginSource>> credentialProvider;
+        private readonly SourceRepository repository;
+        private readonly ILogger logger;
+
+        public static readonly int PageCount = 20;
+
+        public NuGetPluginDiscoverer(UriPluginSource pluginSource) 
+            : this(pluginSource, null)
+        {
+        }
+
+        public NuGetPluginDiscoverer(
+            UriPluginSource pluginSource,
+            Lazy<ICredentialProvider<UriPluginSource>> credentialProvider)
+        {
+            this.pluginSource = pluginSource;
+            this.packageSource = new PackageSource(pluginSource.Uri.ToString());
+            this.credentialProvider = credentialProvider;
+            this.logger = new NullLogger();
```

Comment:
Does not need to be in this PR, but I'm wondering if we should expose a singleton for NullLogger